src/utils/internal.ts

```diff
@@ -710,3 +714,57 @@ export function toHumanReadable<T>(obj: T): HumanReadableType<T> {
 
   return obj as HumanReadableType<T>;
 }
+
+/**
+ * @hidden
+ *
+ * Checks chain version. This function uses a websocket as it's intended to be called during initialization
+ * @param nodeUrl - URL for the chain node
+ * @returns A promise that resolves if the version is in the expected range, otherwise it will reject
+ */
+export function assertExpectedChainVersion(nodeUrl: string): Promise<unknown> {
```

Comment:
I'd recommend doing everything from inside the Promise constructor. It's more readable IMO. If not, at least define the callbacks and create the promise at the top of the function and use `resolve`, `reject`, `_resolve`, `_reject` for names

Also, you're mixing arrow notation with `function` in the event callbacks